modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3011,7 +3011,46 @@ void GDScriptAnalyzer::reduce_assignment(GDScriptParser::AssignmentNode *p_assig
 		GDScriptParser::IdentifierNode *id = static_cast<GDScriptParser::IdentifierNode *>(p_assignment->assignee);
 		// Use == 1 here because this assignment was already counted in the beginning of the function.
 		if (id->source == GDScriptParser::IdentifierNode::LOCAL_VARIABLE && id->variable_source && id->variable_source->assignments == 1) {
-			parser->push_warning(p_assignment, GDScriptWarning::UNASSIGNED_VARIABLE_OP_ASSIGN, id->name);
+			String assignment_op;
+			switch (p_assignment->operation) {
+				case GDScriptParser::AssignmentNode::OP_ADDITION:
+					assignment_op = "+=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_SUBTRACTION:
+					assignment_op = "-=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_MULTIPLICATION:
+					assignment_op = "*=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_DIVISION:
+					assignment_op = "/=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_POWER:
+					assignment_op = "**=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_MODULO:
+					assignment_op = "%=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_BIT_AND:
+					assignment_op = "&=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_BIT_OR:
+					assignment_op = "|=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_BIT_XOR:
+					assignment_op = "^=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_BIT_SHIFT_LEFT:
+					assignment_op = "<<=";
+					break;
+				case GDScriptParser::AssignmentNode::OP_BIT_SHIFT_RIGHT:
+					assignment_op = ">>=";
+					break;
+				default:
+					assignment_op = "";
+					break;
+			}
```

Comment:
There are [`Variant::get_operator_name()`](https://github.com/godotengine/godot/blob/6a6a1168a5a7b14ef988f3516303b6d67a985ebd/core/variant/variant_op.cpp#L1145) and [`AssignmentNode::variant_op`](https://github.com/godotengine/godot/blob/6a6a1168a5a7b14ef988f3516303b6d67a985ebd/modules/gdscript/gdscript_parser.h#L437).

Reply:
Thank you! That is *much* cleaner 😅